lib/puppet/pops/parser/egrammar.ra

```diff
@@ -344,7 +344,12 @@ if_expression
   # Produces Model::IfExpression
   if_part
     : expression LBRACE statements RBRACE else {
-        result = Factory.IF(val[0], Factory.block_or_expression(val[2]), val[4])
+        then_result = Factory.block_or_expression(val[2])
+        if val[2].size > 1
+          loc(then_result, val[1], val[3])
+        end
```

Comment:
The logic in the `loc` method just relays to the `Factory` so it is easy to do this in the `block_or_expression` method.